pylint_nautobot/incorrect_base_class.py

```diff
@@ -1,31 +1,79 @@
-"""Check for imports whose paths have changed in 2.0."""
+"""Check for incorrect base classes."""
+from typing import NamedTuple
+
 from astroid import Assign
 from astroid import ClassDef
 from astroid import Const
 from pylint.checkers import BaseChecker
 
-from pylint_nautobot.utils import is_nautobot_v2_installed
+from .utils import is_version_compatible
 
 
-def is_abstract(node):
+def is_abstract(node: ClassDef) -> bool:
     """Given a node, returns whether it is an abstract base model."""
     for child_node in node.get_children():
         if not (isinstance(child_node, ClassDef) and child_node.name == "Meta"):
             continue
+
         for meta_child in child_node.get_children():
             if (
                 not isinstance(meta_child, Assign)
-                or not meta_child.targets[0].name == "abstract"
+                or not meta_child.targets[0].name == "abstract"  # type: ignore
                 or not isinstance(meta_child.value, Const)
             ):
                 continue
             # At this point we know we are dealing with an assignment to a constant for the 'abstract' field on the
             # 'Meta' class. Therefore, we can assume the value of that to be whether the node is an abstract base model
             # or not.
             return meta_child.value.value
+
     return False
 
 
+_CLASS_MAPPING = (
+    {
+        "incorrect": "django_filters.filters.FilterSet",
+        "correct": "django_filters.filters.BaseFilterSet",
+    },
+    {
+        "incorrect": "django.db.models.base.Model",
+        "correct": "nautobot.core.models.BaseModel",
```

Comment:
Not sure if we support OR but Primary should always be the preferred and base is only used for custom through tables and a few other scenarios.
```suggestion
        "correct": ["nautobot.apps.models.PrimaryModel","nautobot.apps.models.BaseModel"]
```

Reply:
`PrimaryModel` is a `BaseModel` descendant. If we want to allow `BaseModel` to be used, it covers `PrimaryModel` as well.

One simple option is to enforce `PrimaryModel`, instead of `BaseModel`. In such case, using `BaseModel` only would have to be disabled by `# pylint: disable=nb-incorrect-base-class` macro.

Should I split it into two pylint rules?

Reply:
I vote enforce `PrimaryModel`

Reply:
`PrimaryModel` is enforced now.